### ISSUE TITLE
Simplify the DID string spec

### ISSUE THREAD

**Issue by youngjoon-lee** [medibloc/panacea-core#56]
1. Let's remove the network ID `mainnet|testnet` from DIDs.
   - I don't see any strong benefit to have it.
      - It also makes implementation complicated.
   - Many [DID implementers](https://www.w3.org/TR/did-spec-registries/#did-methods) don't have it. (50% vs 50%, roughly)
   - Even, it would be more flexible not to have it.
      - When migrating DIDs from network-1 to network-2, it would be painful to update all DIDs and related VCs. (even though I'm not sure if it might happen.)

2. For `method-specific-id-string` in DIDs, let's do `base58(sha256(pubkey))` instead of `base58(truncate(pubkey, 16))`.
   - The current way `base58(truncate(pubkey, 16))` was inspired from [Sovrin](https://sovrin-foundation.github.io/sovrin/spec/did-method-spec-template.html).
   - But, most DID implementers don't use truncation.
   - I think it would be simpler to do `base58(sha256(pubkey))`. (eg. [Hedera](https://github.com/hashgraph/did-method/blob/master/did-method-specification.md#namespace-specific-identifier-nsi))
   - I also considered having `base58(ripemd160(sha256(pubkey)))` as Bitcoin does, but I don't see any benefit to do `ripemd160` even though it's a legacy hash which isn't [recommended to use](https://pkg.go.dev/golang.org/x/crypto/ripemd160).
   - Of course, `ripemd160` makes DIDs shorter, but it's not a big deal, I guess.

In short,
BEFORE: `did:panacea:mainnet:DnreD8QqXAQaEW9DwC16Wh`
AFTER: `did:panacea:7Prd74ry1Uct87nZqL3ny7aR7Cg46JamVbJgk8azVgUm`
